hub/api/tests/test_api.py

```diff
@@ -63,6 +63,19 @@ def test_populate_dataset(ds):
     assert ds.meta == {"tensors": ["image"]}
 
 
+def test_stringify(memory_ds):
+    ds = memory_ds
+    ds.create_tensor("image")
+    ds.image.extend(np.ones((4, 4)))
+    assert str(ds) == "Dataset(mode='a', tensors=['image'])"
```

Comment:
should also have the url. People might have multiple datasets open and url would be an easy differntiator.

Reply:
@AbhinavTuli Is a storage provider guaranteed to have a `root` field? Or should we store the path in addition to the provider